### ISSUE TITLE
search for text within a note

### ISSUE THREAD

**Issue by odinnaodin** [Sambigeara/fuzzynote#80]
now the search works only by the title, but it is convenient to search for text also inside the note, to search for text throughout the text of the note.

**Comment by Sambigeara** [comment]
Hello!

I've thought about this once or twice in the past. I'm tentative to implement a thorough solution because I envision some changes being made in the future which would alter the underlying mechanisms. There's also considerations around _how_ lines matched via notes will present themselves to the user 🤔

That being said; for the time being, I'd be happy enough to add basic support and allow users to toggle it explicitly via something like `fzn --match-notes` or `$FZN_MATCH_NOTES=true` etc. In terms of how this will present in the UI, it might be a bit misleading given the matched text might not be visible, but at least the user has explicit control over it.

I'll add this to my to-do list to implement as per the above 👍 . I'll be a little tied up in the coming days but should have some time over the weekend.

Cheers

**Comment by odinnaodin** [comment]
> Hello!
> 
> I've thought about this once or twice in the past. I'm tentative to implement a thorough solution because I envision some changes being made in the future which would alter the underlying mechanisms. There's also considerations around _how_ lines matched via notes will present themselves to the user 
> 
> That being said; for the time being, I'd be happy enough to add basic support and allow users to toggle it explicitly via something like `fzn --match-notes` or `$FZN_MATCH_NOTES=true` etc. In terms of how this will present in the UI, it might be a bit misleading given the matched text might not be visible, but at least the user has explicit control over it.
> 
> I'll add this to my to-do list to implement as per the above  . I'll be a little tied up in the coming days but should have some time over the weekend.
> 
> Cheers

I would also like to have a preview panel as it is done for example in https://github.com/alok/notational-fzf-vim

**Comment by Sambigeara** [comment]
> I would also like to have a preview panel as it is done for example in https://github.com/alok/notational-fzf-vim

I agree that this will be a nice feature to have, but drastically increases the size of this task (and probably will require a lot of thought to do it properly/extensibly). I'll tackle the basic note matching support I mentioned above within the scope of this issue.

Cheers
